sql/src/main/java/io/crate/analyze/symbol/InputColumn.java

```diff
@@ -153,6 +141,7 @@ public boolean equals(Object o) {
         InputColumn that = (InputColumn) o;
 
         if (index != that.index) return false;
+        if (!dataType.equals(that.dataType)) return false;
```

Comment:
Without it:
`java.lang.AssertionError: 
Expected: is "2| Rupert\n2| Ford\n1| Earth\n1| Arthur\n"
     but: was "2| 12884901889\n2| 4294967297\n1| 12884901888\n1| 4294967296\n" <Click to see difference>


	at __randomizedtesting.SeedInfo.seed([C70ED727E423FA9E:E913CD112D203A42]:0)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:865)
	at org.junit.Assert.assertThat(Assert.java:832)
	at io.crate.integrationtests.FetchOperationIntegrationTest.testUnionFetch(FetchOperationIntegrationTest.java:118)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1764)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:871)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:907)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:921)`